app/models/manageiq/providers/ibm_power_hmc/infra_manager/provision/cloning.rb

```diff
@@ -3,29 +3,56 @@ def clone_complete?
     $ibm_power_hmc_log.info("#{self.class}##{__method__}")
     raise MiqException::MiqProvisionError, "VM capture to template failed" if phase_context[:new_vm_ems_ref].nil?
 
+    association = source.template? ? :vms : :miq_templates
     target = InventoryRefresh::Target.new(:manager     => source.ext_management_system,
-                                          :association => :miq_templates,
+                                          :association => association,
                                           :manager_ref => {:ems_ref => phase_context[:new_vm_ems_ref]})
     EmsRefresh.queue_refresh(target)
     true
   end
 
   def find_destination_in_vmdb(ems_ref)
     $ibm_power_hmc_log.info("#{self.class}##{__method__}")
-    ManageIQ::Providers::IbmPowerHmc::InfraManager::Template.find_by(:name => dest_name, :ems_ref => ems_ref)
+    if source.template?
+      ManageIQ::Providers::IbmPowerHmc::InfraManager::Lpar.find_by(:name => dest_name, :ems_ref => ems_ref)
+    else
+      ManageIQ::Providers::IbmPowerHmc::InfraManager::Template.find_by(:name => dest_name, :ems_ref => ems_ref)
+    end
```

Comment:
I see this was like this before, but we really need to be including the ems_id here.  You can either add it to the `find_by` like `ManageIQ::Providers::IbmPowerHmc::InfraManager::Lpar.find_by(:ext_management_system => source.ext_management_system, :name => dest_name, :ems_ref => ems_ref)` or use the association like `source.ext_management_system.vms.find_by(:name => dest_name, :ems_ref => ems_ref)`

Also if the ems_ref is unique (it needs to be) then you don't need the name in the query.